### PR TITLE
[MPICH] Fix musl build

### PR DESCRIPTION
**Pull request by simonbyrne**
See https://github.com/JuliaPackaging/Yggdrasil/pull/313#issuecomment-575337293

### CODE REVIEW

File: M/MPICH/build_tarballs.jl [hunk 1]
```diff
@@ -20,10 +20,6 @@ make "${flags[@]}" -j${nproc}
 make "${flags[@]}" install
 """
 
-# Windows and MUSL are not supported
```
Comment (giordano) [inline M/MPICH/build_tarballs.jl]: Probably you wanted to keep
```julia
platforms = supported_platforms()
```

**Comment by simonbyrne** [comment]
i686 MUSL failure:
```
[23:15:13] checking size of bool... 0
[23:15:13] configure: error: unable to determine matching C type for C++ bool
[23:15:13]  ---> ./configure --prefix=$prefix --host=$target --enable-shared=yes --enable-static=no --disable-dependency-tracking --disable-fortran --docdir=/tmp --enable-timer-type=gettimeofday
[23:15:13] Previous command exited with 1
```

Windows failures:
```
[23:15:57] configure: error: cannot support shared memory:  need either sysv shared memory functions or mmap in order to support shared memory
[23:15:57] configure: error: MPL configure failed
[23:15:57]  ---> ./configure --prefix=$prefix --host=$target --enable-shared=yes --enable-static=no --disable-dependency-tracking --disable-fortran --docdir=/tmp --enable-timer-type=gettimeofday
[23:15:57] Previous command exited with 1
[23:15:57] 
```

**Comment by ViralBShah** [comment]
It is unclear if shared memory support can be disabled to see if it will build on windows.

**Comment by ViralBShah** [comment]
Perhaps best to install msmpi for Windows. 

https://github.com/Microsoft/Microsoft-MPI

**Comment by simonbyrne** [comment]
I agree. Any idea what is going on with Musl though?

**Comment by giordano** [comment]
> Any idea what is going on with Musl though?

The test that is failing ("determine matching C type for C++ bool") is trying to run an executable.  There is a problem with our `i686-linux-musl` system: theoretically we should able to run C++ programs for this platform in the BB environment, but in practice there is something broken that prevents us from doing so.  I had found the same problem in https://github.com/JuliaPackaging/Yggdrasil/pull/91#discussion_r326861961.

We can workaround this issue by passing a cached value of `ac_cv_sizeof_bool`

**Comment by giordano** [comment]
For `i686-linux-gnu` the value is
```sh
ac_cv_sizeof_bool='1'
```
Assuming that it would be the same for `i686-linux-musl`,
```sh
./configure --prefix=$prefix --host=$target --enable-shared=yes --enable-static=no --disable-dependency-tracking --disable-fortran --docdir=/tmp --enable-timer-type=gettimeofday ac_cv_sizeof_bool='1'
```
goes through

**Comment by staticfloat** [comment]
From what I'm reading, it looks like [mingw just doesn't have the shared memory support](https://lists.mcs.anl.gov/pipermail/mpich-discuss/2010-November/008486.html) that "nemesis" wants.  I wonder if we could disable nemesis channels on windows?  I don't know if that is useful or not.

Their suggestion is to either use Cygwin, or use the Visual Studio-based binaries they serve on their website, which they say "should be mingw compatible".

**Comment by simonbyrne** [comment]
I think we leave Windows for now.

**Comment by simonbyrne** [comment]
> Their suggestion is to either use Cygwin, or use the Visual Studio-based binaries they serve on their website, which they say "should be mingw compatible".

I haven't seen any examples of building MPICH on Windows, so I think MS MPI (which is an MPICH derivative) is the way to go.

**Pull request merged**